ui-manchette/src/components/OperationalPointList.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect } from 'react';
+import { OperationalPointType } from '../types/pathPropertiesTypes';
+import OperationalPoint from './OperationalPoint';
+import cx from 'classnames';
+import { positionKm } from './helpers';
+
+interface OperationalPointListProps {
+  operationalPoints: OperationalPointType[];
+  isProportionnal?: boolean;
+  zoom?: number;
+  width?: number;
+}
+
+const OperationalPointList: React.FC<OperationalPointListProps> = ({
+  operationalPoints,
+  isProportionnal = false,
+  zoom = 1,
+}) => {
+  const [operationalPointsToDisplay, setOperationalPointsToDisplay] = React.useState<
+    OperationalPointType[]
+  >([]);
+  const baseHeight = 32;
+
+  const operationalPointStyle = (op: OperationalPointType, nextOp: OperationalPointType) => {
+    if (isProportionnal) {
+      if (!nextOp) {
+        return { height: `${baseHeight}px` };
+      } else {
+        return { height: `${positionKm(nextOp.position - op.position) * zoom * 8}px` };
+      }
+    } else {
+      return { height: `${baseHeight * zoom}px` };
+    }
+  };
+
+  useEffect(() => {
+    const calcOperationalPointsToDisplay = () => {
+      if (isProportionnal) {
+        const result = operationalPoints.reduce(
```

Comment:
LGTM! For improved readability and consistency, maybe some variables might need renaming: `op: lastDisplayedPoint, diff: distance, acc: accumulatedPoints, nextopt: currentPoint, lastElement: lastPoint for consistency `